src/storage/models.rs

```diff
@@ -68,27 +101,8 @@ pub struct ListSnippetsQuery {
     /// If set, only the snippets that have *one of* the specified tags attached
     /// will be returned.
     pub tags: Option<Vec<String>>,
-    /// If set, up to this number of snippets will be returned.
-    pub limit: Option<usize>,
-    /// If set, this number of snippets will be skipped (snippets are sorted in
-    /// the reverse chronological order) before the limit is applied.
-    pub offset: Option<usize>,
-}
-
-impl Default for ListSnippetsQuery {
-    fn default() -> Self {
-        // default filters should match all snippets. The only constraint we want
-        // to enforce is the maximum number of snippets in the response. Here it
-        // is set to a fairly small value; API users are expected to use pagination
-        // to retrieve more snippets if needed.
-        ListSnippetsQuery {
-            title: None,
-            syntax: None,
-            tags: None,
-            limit: Some(DEFAULT_LIMIT_SIZE),
-            offset: None,
-        }
-    }
```

Comment:
So if it wasn't for `limit`, I could just derive `Default` and that would indeed automatically call `default()` for each attribute.  However, because I want to slightly customize the implementation, that does not work for us :(